wagtail/images/models.py

```diff
@@ -1055,6 +1056,20 @@ def run(self, image: AbstractImage, output: BytesIO, source: File = None):
                 else:
                     quality = getattr(settings, "WAGTAILIMAGES_AVIF_QUALITY", 80)
                 return willow.save_as_avif(output, quality=quality)
+            elif output_format == "heic":
+                # Allow changing of HEIC compression quality. Safari is the only browser that supports HEIC,
+                # so there is little value in outputting it - for that reason, we make it work if someone
+                # explicitly requests it, but these settings are not documented.
+                if (
+                    "output-format-options" in env
+                    and "lossless" in env["output-format-options"]
+                ):
+                    return willow.save_as_heif(output, lossless=True)
+                elif "heic-quality" in env:
+                    quality = env["heic-quality"]
+                else:
+                    quality = getattr(settings, "WAGTAILIMAGES_HEIC_QUALITY", 80)
+                return willow.save_as_heif(output, quality=quality)
```

Comment:
I got an attribute error because Willow's method name is actually `save_as_heic`:

https://willow.wagtail.org/latest/guide/save.html#lossless-avif-heic-and-webp

```suggestion
                    return willow.save_as_heic(output, lossless=True)
                elif "heic-quality" in env:
                    quality = env["heic-quality"]
                else:
                    quality = getattr(settings, "WAGTAILIMAGES_HEIC_QUALITY", 80)
                return willow.save_as_heic(output, quality=quality)
```

I also had to add `heic` to the following list in order for `format-heic` to work, otherwise I got `InvalidFilterSpecError`:

https://github.com/wagtail/wagtail/blob/28fcd01a31374e15d4e40e6f42f4f81445b85f77/wagtail/images/image_operations.py#L411-L412

Reply:
Good catch! Fixed (with test added) in 15abe65eb4fcf2f4610a3b5d99784e55f672e71d.